src/language/parser.ts

```diff
@@ -435,9 +443,77 @@ export class Parser {
       selectionSet: this.peek(TokenKind.BRACE_L)
         ? this.parseSelectionSet()
         : undefined,
+      required: this.parseRequiredStatus(),
```

Comment:
Shouldn't this come before the selection set and before the directives?

`Field : Alias? Name Arguments? Nullability? Directives? SelectionSet?`


The kitchen sink test never tests a field with selection set:

```
      field3!
       requiredField4: field4!
       field5?
       optionalField6: field6?
       unsetListItemsRequiredList: listField[]!
       requiredListItemsUnsetList: listField[!]
       requiredListItemsRequiredList: listField[!]!
       unsetListItemsOptionalList: listField[]?
       optionalListItemsUnsetList: listField[?]
       optionalListItemsOptionalList: listField[?]?
       multidimensionalList: listField[[[!]!]!]!
```

This must also be tested in its various combinations
```
{
  field? {
    name
  }
}
```

Reply:
Yeah good point. I'll put up some more tests

Reply:
I have fixed this issue in our implementation and have tests now for this.

It must be like the following:

```
return this.node<FieldNode>(start, {
      kind: Kind.FIELD,
      alias,
      name,
      arguments: this.parseArguments(false),
      required: this.parseRequiredStatus(),
      directives: this.parseDirectives(false),
       selectionSet: this.peek(TokenKind.BRACE_L)
         ? this.parseSelectionSet()
         : undefined
```
       
At the moment it would break in your implementation when it is combined with a selection set or if you combine tests with directives.


Here is our implementation of this ... its c# but the order is correct.
https://github.com/ChilliCream/hotchocolate/blob/b1243030b9e20f6c4a22780857e5cd2031d16c68/src/HotChocolate/Language/src/Language.Utf8/Utf8GraphQLParser.Operations.cs#L255

Reply:
Yep! That's what I went with too

https://github.com/graphql/graphql-js/pull/3418/files#diff-297a41e54c25bbd5c846bb55d7acaaca9f2c9f5fa654b19004da4efbf9a88154R463